logos/logos-workernode/logos_worker_node/lane_manager.py

```
@@ -1439,8 +1437,10 @@ async def _auto_place_gpu_devices(self, lane_id: str, lane_config: LaneConfig) -
             raise RuntimeError(
                 f"Auto-placement: no feasible GPU subset for lane '{lane_id}' "
                 f"model={lane_config.model} (required≈{required_total_mb:.0f}MB total, "
-                f"tp={tp_size}, per-GPU need≈{per_gpu_required_mb:.0f}MB+"
-                f"{headroom_mb:.0f}MB headroom; per-GPU free: {per_gpu_summary})"
+                f"tp={tp_size}, per-GPU threshold≈{per_gpu_threshold_mb:.0f}MB "
+                f"(estimate {per_gpu_required_mb:.0f}MB × {_GPU_PLACEMENT_SECURITY_RATIO:.3f} "
```

Ruff (0.15.14)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)